experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -603,37 +568,36 @@ const mapPrototype = Object.create(Object.prototype, mapStaticDispatchMap);
 
 // #endregion
 
-const getMapNode = <TSchema extends MapSchema>(target: object): MapNode<TSchema> => {
-	return getTreeNode(target) as MapNode<TSchema>;
-};
-
-function createMapProxy<TSchema extends MapSchema>(treeNode: TreeNode): SharedTreeMap<TSchema> {
+function createMapProxy<TSchema extends MapSchema>(): SharedTreeMap<TSchema> {
 	// Create a 'dispatch' object that this Proxy forwards to instead of the proxy target.
-	const dispatch = Object.create(mapPrototype, {
+	const dispatch: object = Object.create(mapPrototype, {
 		// Empty - JavaScript Maps do not expose any "own" properties.
 	});
```

Comment:
@noencke  - You should go ahead with your change, but just as an FYI - one of the things I've been wanting to benchmark is if creation is faster if we avoid Object.create() followed by Object.defineProperty() by doing something like this:

```ts
const dispatch = Object.create(mapPrototype, {
    [treeNodeSym]: treeNode
});
```

Reply:
That's a cool idea. I think there are a lot of opportunities for micro-ops in this file! We could also benchmark using a second WeakMap for the mapping. I read something online that said WeakMap lookups are slower than symbol reads, but then I read something else more recent that says V8 implements WeakMaps by just putting a hidden field on the target object, so it's more or less the same. We'll have to do our own analysis to see.